lib/util/ios-proxy.js

```diff
@@ -148,13 +149,53 @@ let iOSProxy = {
     unload(() => proxyProcess && proxyProcess.kill());
   },
 
-  checkIfBound() {
-    return new Promise(resolve => {
-      console.log("Checking if proxy is bound to port 9230");
-      let cmd = "/usr/sbin/lsof -i:9230";
-      child_process.exec(cmd, err => resolve(!err));
-    });
-  },
+  checkIfBound: task.async(function*() {
+    console.log("Checking if proxy is bound to port 9230");
+    let args = [], command = "lsof -i:9230";
+    // Run the command through a shell command in order to support non absolute
+    // paths.
+    // On Windows the `ComSpec` env variable is going to refer to cmd.exe. On
+    // Linux and Mac, the SHELL env variable should refer to the user-chosen
+    // shell program.
+    // We do not check for OS, as on Windows, with Cygwin, ComSpec isn't set.
+    let envService = Cc["@mozilla.org/process/environment;1"].getService(Ci.nsIEnvironment);
+    let shell = envService.get("ComSpec") || envService.get("SHELL");
+    args.unshift(command);
+
+    // For cmd.exe, we have to pass the `/C` option,
+    // but for unix shells we need -c.
+    // That to interpret next argument as a shell command.
+    if (envService.exists("ComSpec")) {
+      args.unshift("/C");
+    } else {
+      args.unshift("-c");
+    }
+
+    let result = false;
+    try {
+      result = yield new Promise(resolve => {
+        Subprocess.call({
+          command: shell,
+          arguments: args,
+          environment: [ "PATH=$PATH:/usr/sbin:/usr/bin" ],
```

Comment:
`/usr/sbin` really should be in the default set for OS X! For example, it's listed in `/etc/paths` on 10.9, which is one of the files OS X considers when building the default PATH set.

I've heard Subprocess ignores all env vars by default, so maybe if you forward the value of PATH from `envService`, you would not need to append specific values.